src/main/java/javax/validation/metadata/ContainerDescriptor.java

```diff
@@ -0,0 +1,35 @@
+package javax.validation.metadata;
+
+import java.util.Set;
+
+import javax.validation.Valid;
+
+/**
+ * Represents an element that might be a container, thus allowing container element types constraints.
+ *
+ * @author Guillaume Smet
+ * @since 2.0
+ */
+public interface ContainerDescriptor {
+
+	/**
+	 * Returns the container element type descriptor for a given type argument.
+	 * <p>
+	 * Returns {@code null} if the container element type does not exist or has no constraint nor is marked as cascaded
+	 * (see {@link #getConstrainedContainerElementTypes()})
+	 *
+	 * @param typeArgumentIndex type argument index evaluated
+	 * @return the container element type descriptor
+	 */
+	ContainerElementTypeDescriptor getConstraintsForContainerElementType(Integer typeArgumentIndex);
```

Comment:
It was made to be able to shoe in the array case. But maybe we should make it an `int` and add another method, should we support the array case in the future?